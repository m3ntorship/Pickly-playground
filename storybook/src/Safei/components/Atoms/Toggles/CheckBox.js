import React, { useState } from "react";

const CheckBox = (props) => {
  const [hideIcon, setHideIcon] = useState(true);

  const [displayIcon, setDisplayIcon] = useState({
    icon: "hidden",
    bg: "bg-white",
  });
  let disablCheck = props.disabled ? "none" : "";
  let opa = props.disabled ? "25" : "";
  const handleClick = () => {
    console.log(hideIcon);
    setHideIcon(!hideIcon);
    if (hideIcon) {
      setDisplayIcon({
        icon: "",
        bg: "bg-primary",
      });
    } else {
      setDisplayIcon({
        icon: "hidden",
        bg: "bg-white",
      });
    }
  };
  let size = "1";
  if (props.size === "sm") {
    size = "0.666";
  } else {
    size = "1";
  }

  return (
    <div
      style={{
        pointerEvents: `${disablCheck}`,
        transform: `scale(${size})`,
      }}
      className={`border-2 border-grey-shd4 w-6 h-6  ${displayIcon.bg} relative cursor-pointer rounded-sm hover:border-grey-shd1 opacity-${opa}`}
      id="box"
      onClick={handleClick}
    >
      <svg
        className={`mt-1.5 ml-1 ${displayIcon.icon}`}
        width="11"
        height="9"
        viewBox="0 0 11 9"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.66632 6.39073L1.47099 4.19539L0.52832 5.13806L3.66632 8.27606L10.1377 1.80473L9.19499 0.862061L3.66632 6.39073Z"
          fill="white"
        />
      </svg>

      <input
        type="checkbox"
        className="h-5 w-5 opacity-0"
        disabled={props.disabled}
      />
    </div>
  );
};
export default CheckBox;
