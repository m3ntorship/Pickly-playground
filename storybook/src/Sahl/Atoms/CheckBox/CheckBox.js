import React, { useState } from "react";
import PropTypes from "prop-types";
import CheckIcon from "../../assets/icons/Check";

export default function CheckBox(props) {
  const { size = "default", disabled = false } = props;
  const [checked, setChecked] = useState(false);

  const onChange = () => {
    setChecked(!checked);
  };
  return (
    <div
      className={`relative flex justify-center items-center ${
        size === "default" ? "w-6 h-6 rounded-sm" : "w-4 h-4 rounded-sm"
      } border border-grey-shd4 hover:border-grey-shd1 ${
        checked && "bg-primary hover:bg-primary-shd8"
      } ${disabled && "opacity-30"} focus:outline-none focus:border-2`}
    >
      <input
        disabled={disabled}
        type="checkbox"
        onChange={onChange}
        className="absolute opacity-0 cursor-pointer w-full h-full"
      />
      {checked && (
        <CheckIcon
          width={size === "default" ? "w-4" : "w-3"}
          height={size === "default" ? "h-4" : "h-3"}
          color="#FFF"
        />
      )}
    </div>
  );
}

CheckBox.propTypes = {
  size: PropTypes.oneOf(["default", "small"]),
  disabled: PropTypes.bool,
};
