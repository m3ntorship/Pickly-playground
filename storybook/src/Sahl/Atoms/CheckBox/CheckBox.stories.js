import React from "react";

import CheckBox from "./CheckBox";

const defaultExport = {
  title: "Sahl/Atoms/CheckBox",
  component: CheckBox,
};

export default defaultExport;

const Template = (args) => <CheckBox {...args} />;

export const Default = Template.bind({});
Default.args = {
  disabled: false,
  size: "default",
};
