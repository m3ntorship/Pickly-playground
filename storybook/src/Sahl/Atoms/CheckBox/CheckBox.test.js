import React from "react";
import renderer from "react-test-renderer";
import CheckBox from "./CheckBox";

describe("CheckBox component", () => {
  it("should render without passing any props", () => {
    const tree = renderer.create(<CheckBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be disabled when passing disabled", () => {
    const tree = renderer.create(<CheckBox disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should be rendered with large size no size is passed", () => {
    const tree = renderer.create(<CheckBox disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should be rendered with large size when size=small", () => {
    const tree = renderer.create(<CheckBox disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
