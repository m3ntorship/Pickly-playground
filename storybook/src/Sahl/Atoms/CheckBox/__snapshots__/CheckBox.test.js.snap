// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component should be disabled when passing disabled 1`] = `
<div
  className="relative flex justify-center items-center w-6 h-6 rounded-sm border border-grey-shd4 hover:border-grey-shd1 false opacity-30 focus:outline-none focus:border-2"
>
  <input
    className="absolute opacity-0 cursor-pointer w-full h-full"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`CheckBox component should be rendered with large size no size is passed 1`] = `
<div
  className="relative flex justify-center items-center w-6 h-6 rounded-sm border border-grey-shd4 hover:border-grey-shd1 false opacity-30 focus:outline-none focus:border-2"
>
  <input
    className="absolute opacity-0 cursor-pointer w-full h-full"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`CheckBox component should be rendered with large size when size=small 1`] = `
<div
  className="relative flex justify-center items-center w-6 h-6 rounded-sm border border-grey-shd4 hover:border-grey-shd1 false opacity-30 focus:outline-none focus:border-2"
>
  <input
    className="absolute opacity-0 cursor-pointer w-full h-full"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`CheckBox component should render without passing any props 1`] = `
<div
  className="relative flex justify-center items-center w-6 h-6 rounded-sm border border-grey-shd4 hover:border-grey-shd1 false false focus:outline-none focus:border-2"
>
  <input
    className="absolute opacity-0 cursor-pointer w-full h-full"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
